diff --git a/node_modules/@verida/client-rn/dist/src/context/datastore.d.ts b/node_modules/@verida/client-rn/dist/src/context/datastore.d.ts
index cfa87b2..71bee4a 100644
--- a/node_modules/@verida/client-rn/dist/src/context/datastore.d.ts
+++ b/node_modules/@verida/client-rn/dist/src/context/datastore.d.ts
@@ -24,7 +24,7 @@ declare class Datastore implements IDatastore {
      *
      * The errors remain until they are replaced by any new errors.
      */
-    errors: object;
+    errors: Array<Error>;
     /**
      * Create a new Datastore.
      *
@@ -83,7 +83,7 @@ declare class Datastore implements IDatastore {
      * console.log(results);
      * @returns {object[]} An array of database records.
      */
-    getMany(customFilter?: any, options?: any): Promise<object[]>;
+    getMany<T extends object = object>(customFilter?: any, options?: any): Promise<T[]>;
     /**
      * Get a single database record that matches.
      *
diff --git a/node_modules/@verida/client-rn/src/context/datastore.ts b/node_modules/@verida/client-rn/src/context/datastore.ts
index 7d09279..e28ceda 100644
--- a/node_modules/@verida/client-rn/src/context/datastore.ts
+++ b/node_modules/@verida/client-rn/src/context/datastore.ts
@@ -31,7 +31,7 @@ class Datastore implements IDatastore {
    *
    * The errors remain until they are replaced by any new errors.
    */
-  public errors: object = {};
+  public errors: Array<Error> = {};
 
   /**
    * Create a new Datastore.
@@ -116,10 +116,10 @@ class Datastore implements IDatastore {
    * console.log(results);
    * @returns {object[]} An array of database records.
    */
-  public async getMany(
+  public async getMany<T extends object = object>(
     customFilter: any = {},
     options: any = {}
-  ): Promise<object[]> {
+  ): Promise<T[]> {
     await this.init();
 
     let filter = _.merge(
