diff --git a/node_modules/@verida/wallet-utils/dist/utils.js b/node_modules/@verida/wallet-utils/dist/utils.js
index c1f8205..bbe7056 100644
--- a/node_modules/@verida/wallet-utils/dist/utils.js
+++ b/node_modules/@verida/wallet-utils/dist/utils.js
@@ -55,8 +55,10 @@ var utils = /** @class */ (function () {
      * @param chain `ethereum` or `vechain`
      */
     utils.createWallet = function (chain) {
+        const time = Date.now();
         var wallet = chains[chain].createWallet();
         var did = "did:" + chain + ":" + wallet["address"];
+        console.log('createWallet:', chain, Date.now() - time)
         return {
             chain: chain,
             mnemonic: wallet["mnemonic"],
@@ -67,8 +69,10 @@ var utils = /** @class */ (function () {
         };
     };
     utils.getWallet = function (chain, mnemonic) {
+        const time = Date.now();
         var wallet = chains[chain].getWallet(mnemonic);
         var did = "did:" + chain + ":" + wallet["address"];
+        console.log('getWallet:', chain, Date.now() - time)
         return {
             chain: chain,
             mnemonic: wallet["mnemonic"],