diff --git a/node_modules/@verida/wallet-utils/node_modules/pbkdf2/lib/async.js b/node_modules/@verida/wallet-utils/node_modules/pbkdf2/lib/async.js
index 3af2c49..9f08ab2 100644
--- a/node_modules/@verida/wallet-utils/node_modules/pbkdf2/lib/async.js
+++ b/node_modules/@verida/wallet-utils/node_modules/pbkdf2/lib/async.js
@@ -1,5 +1,11 @@
 var Buffer = require('safe-buffer').Buffer
 
+var Crypto = require('react-native-quick-crypto')
+
+var bytes_1 = require("@ethersproject/bytes");
+var sha2_1 = require("@ethersproject/sha2");
+var base64 = require('@ethersproject/base64')
+
 var checkParameters = require('./precondition')
 var defaultEncoding = require('./default-encoding')
 var sync = require('./sync')
@@ -84,35 +90,16 @@ function resolvePromise (promise, callback) {
   })
 }
 module.exports = function (password, salt, iterations, keylen, digest, callback) {
-  if (typeof digest === 'function') {
-    callback = digest
-    digest = undefined
-  }
-
-  digest = digest || 'sha1'
-  var algo = toBrowser[digest.toLowerCase()]
+  // Fallback to the pbkdf2Sync function (so that it's compatible with the @ethersproject/pbkdf2 module)
+  if (!callback) return Crypto.pbkdf2Sync(password, salt, iterations, keylen, digest)
 
-  if (!algo || typeof global.Promise !== 'function') {
-    getNextTick()(function () {
-      var out
-      try {
-        out = sync(password, salt, iterations, keylen, digest)
-      } catch (e) {
-        return callback(e)
-      }
-      callback(null, out)
-    })
-    return
-  }
-
-  checkParameters(iterations, keylen)
-  password = toBuffer(password, defaultEncoding, 'Password')
-  salt = toBuffer(salt, defaultEncoding, 'Salt')
-  if (typeof callback !== 'function') throw new Error('No callback provided to pbkdf2')
-
-  resolvePromise(checkNative(algo).then(function (resp) {
-    if (resp) return browserPbkdf2(password, salt, iterations, keylen, algo)
-
-    return sync(password, salt, iterations, keylen, digest)
+  resolvePromise(new Promise((resolve, reject) => {
+    try {
+      const res = Crypto.pbkdf2Sync(password, salt, iterations, keylen, digest)
+      resolve(res)
+    } catch (error) {
+      reject(error)
+    }
   }), callback)
+
 }
diff --git a/node_modules/@verida/wallet-utils/node_modules/pbkdf2/lib/default-encoding.js b/node_modules/@verida/wallet-utils/node_modules/pbkdf2/lib/default-encoding.js
index 3042099..35e04d5 100644
--- a/node_modules/@verida/wallet-utils/node_modules/pbkdf2/lib/default-encoding.js
+++ b/node_modules/@verida/wallet-utils/node_modules/pbkdf2/lib/default-encoding.js
@@ -2,8 +2,8 @@ var defaultEncoding
 /* istanbul ignore next */
 if (global.process && global.process.browser) {
   defaultEncoding = 'utf-8'
-} else if (global.process && global.process.version) {
-  var pVersionMajor = parseInt(process.version.split('.')[0].slice(1), 10)
+} else if (global.process && "v18.16.0") {
+  var pVersionMajor = parseInt("v18.16.0".split('.')[0].slice(1), 10)
 
   defaultEncoding = pVersionMajor >= 6 ? 'utf-8' : 'binary'
 } else {
diff --git a/node_modules/@verida/wallet-utils/node_modules/pbkdf2/lib/sync-browser.js b/node_modules/@verida/wallet-utils/node_modules/pbkdf2/lib/sync-browser.js
index 20a64b4..594e366 100644
--- a/node_modules/@verida/wallet-utils/node_modules/pbkdf2/lib/sync-browser.js
+++ b/node_modules/@verida/wallet-utils/node_modules/pbkdf2/lib/sync-browser.js
@@ -2,6 +2,10 @@ var md5 = require('create-hash/md5')
 var RIPEMD160 = require('ripemd160')
 var sha = require('sha.js')
 var Buffer = require('safe-buffer').Buffer
+var fastCrypto = require('react-native-fast-crypto')
+var pbkdf2 = fastCrypto.pbkdf2
+
+var Crypto = require('react-native-quick-crypto')
 
 var checkParameters = require('./precondition')
 var defaultEncoding = require('./default-encoding')
@@ -67,39 +71,14 @@ function getDigest (alg) {
   return shaFunc
 }
 
-function pbkdf2 (password, salt, iterations, keylen, digest) {
-  checkParameters(iterations, keylen)
-  password = toBuffer(password, defaultEncoding, 'Password')
-  salt = toBuffer(salt, defaultEncoding, 'Salt')
-
-  digest = digest || 'sha1'
-
-  var hmac = new Hmac(digest, password, salt.length)
-
-  var DK = Buffer.allocUnsafe(keylen)
-  var block1 = Buffer.allocUnsafe(salt.length + 4)
-  salt.copy(block1, 0, 0, salt.length)
-
-  var destPos = 0
-  var hLen = sizes[digest]
-  var l = Math.ceil(keylen / hLen)
-
-  for (var i = 1; i <= l; i++) {
-    block1.writeUInt32BE(i, salt.length)
-
-    var T = hmac.run(block1, hmac.ipad1)
-    var U = T
-
-    for (var j = 1; j < iterations; j++) {
-      U = hmac.run(U, hmac.ipad2)
-      for (var k = 0; k < hLen; k++) T[k] ^= U[k]
-    }
-
-    T.copy(DK, destPos)
-    destPos += hLen
-  }
-
-  return DK
+function jspbkdf2 (password, salt, iterations, keylen, digest) {
+  return Crypto.pbkdf2Sync(
+    password,
+    salt,
+    iterations,
+    keylen,
+    digest
+  )
 }
 
-module.exports = pbkdf2
+module.exports = jspbkdf2
