diff --git a/node_modules/@verida/wallet-utils/node_modules/near-seed-phrase/index.js b/node_modules/@verida/wallet-utils/node_modules/near-seed-phrase/index.js
index 9c24421..da5152f 100644
--- a/node_modules/@verida/wallet-utils/node_modules/near-seed-phrase/index.js
+++ b/node_modules/@verida/wallet-utils/node_modules/near-seed-phrase/index.js
@@ -13,7 +13,8 @@ const normalizeSeedPhrase = (seedPhrase) => seedPhrase.trim().split(/\s+/).map(p
 
 const parseSeedPhrase = (seedPhrase, derivationPath) => {
     const seed = bip39.mnemonicToSeed(normalizeSeedPhrase(seedPhrase))
-    const { key } = derivePath(derivationPath || KEY_DERIVATION_PATH, seed.toString('hex'))
+    const seedString = typeof seed === 'string' ? seed.substring(2,seed.length) : seed.toString('hex')
+    const { key } = derivePath(derivationPath || KEY_DERIVATION_PATH, seedString)
     const keyPair = nacl.sign.keyPair.fromSeed(key)
     const publicKey = 'ed25519:' + bs58.encode(Buffer.from(keyPair.publicKey))
     const secretKey = 'ed25519:' + bs58.encode(Buffer.from(keyPair.secretKey))
