diff --git a/node_modules/react-native-sqlite-2/ios/RNSqlite2.h b/node_modules/react-native-sqlite-2/ios/RNSqlite2.h
index bcaa915..3ba1d4c 100644
--- a/node_modules/react-native-sqlite-2/ios/RNSqlite2.h
+++ b/node_modules/react-native-sqlite-2/ios/RNSqlite2.h
@@ -1,10 +1,4 @@
-
-#if __has_include("RCTBridgeModule.h")
-#import "RCTBridgeModule.h"
-#else
 #import <React/RCTBridgeModule.h>
-#endif
-
 #import <Foundation/Foundation.h>
 
 #ifdef DEBUG_RNSqlite2
