diff --git a/node_modules/react-native-code-push/android/app/build.gradle b/node_modules/react-native-code-push/android/app/build.gradle
index 4fae97e..3e9b7f5 100644
--- a/node_modules/react-native-code-push/android/app/build.gradle
+++ b/node_modules/react-native-code-push/android/app/build.gradle
@@ -27,5 +27,5 @@ android {
 
 dependencies {
     implementation "com.facebook.react:react-native:+"
-    implementation 'com.nimbusds:nimbus-jose-jwt:5.1'
+    implementation 'com.nimbusds:nimbus-jose-jwt:8.0'
 }
